core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -1853,12 +1853,14 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
   /**
    * The <code>FLOOR</code> function.
    */
-  public static final SqlFunction FLOOR = new SqlFloorFunction(SqlKind.FLOOR);
+  public static final SqlFunction FLOOR =
+      new SqlFloorFunction(SqlKind.FLOOR, ReturnTypes.ARG0_OR_EXACT_NO_SCALE);
```

Comment:
I would keep the same constructor, and add methods `withName`, `withReturnTypeInference`. (We have discovered that for functions/operators, subclassing doesn't scale, and adding constructor parameters doesn't scale.)